packages/gamut/src/ConnectedForm/ConnectedForm.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useMemo } from 'react';
+import {
+  FormProvider,
+  FormProviderProps,
+  Mode,
+  RegisterOptions,
+  SubmitHandler,
+  useForm,
+  UseFormOptions,
+} from 'react-hook-form';
+
+import { Form } from '../Form';
+import { FormProps } from '../Form/Form';
+import { FormValues } from '../Form/types';
+import { submitSuccessStatus } from './utils';
+
+export interface FormContextProps {
+  /**
+   * If fields should be disabled while form is being submitted and after successful submission.
+   */
+  disableFieldsOnSubmit?: boolean;
+  /**
+   * If fields should be reset after successful submission.
+   */
+  resetOnSubmit?: boolean;
+  /**
+   * If required field should show asterisks in labels.
+   */
+  showRequired?: boolean;
+  /**
+   * Function called with field values on submit, if all validations have passed.
+   */
+  validationRules?: { string?: RegisterOptions };
+  /**
+   * Sets if form submission was successful - if `undefined` will fall back to react-hook-forms native formState.isSubmitSuccessful.
+   */
+  wasSubmitSuccessful?: boolean;
+}
+
+export interface ConnectedFormProps<Values extends {}>
+  extends Omit<FormContextProps, 'validationRules'>,
+    Omit<FormProps, 'onSubmit'> {
+  children?: React.ReactNode;
+
+  /**
+   * Function called with field values on submit, if all validations have passed.
+   */
+  onSubmit: SubmitHandler<Values>;
+
+  /**
+   * Default values are highly recommended for reliable form behavior, particularly resets.
+   */
+  defaultValues?: UseFormOptions<Values>['defaultValues'];
+
+  /**
+   * Function called with field values on submit, if all validations have passed.
+   */
+  validationRules?: Partial<{ [Key in keyof Values]?: RegisterOptions }>;
```

Comment:
Nit: having both `Partial` and the `?` is redundant, I think?

```suggestion
  validationRules?: { [Key in keyof Values]?: RegisterOptions }>
```
```suggestion
  validationRules?: Partial<{Record<keyof Values, RegisterOptions>>;
```